src/commands/deployments/editScmType.ts

```diff
@@ -18,7 +18,14 @@ export async function editScmType(context: IActionContext, node?: SiteTreeItem |
         node = <SiteTreeItem>node.parent;
     }
 
-    await appservice.editScmType(node.root.client, node, context, newScmType, showToast);
+    if (node instanceof DeploymentsTreeItem && node.parent instanceof AzureParentTreeItem) {
```

Comment:
You only need to check if the node is an instanceof AzureParentTreeItem (or just SiteTreeItem alternatively) since we check if it's a deploymentsTreeItem in L15:18